src/dashboard/dashboard.js

```diff
@@ -1,10 +1,77 @@
 import React from "react";
+import ChatListComponent from "../chatlist/chatList";
+import { Button, withStyles } from "@material-ui/core";
+import styles from "./styles";
+import ChatViewComponent from "../chatview/chatView";
+import firebase from "../firebase";
 
 class DashboardComponent extends React.Component {
-    
-    render() {
-        return(<div>Hello World from Dashboard</div>)
-    }
+  constructor() {
+    super();
+    this.state = {
+      selectedChat: null,
+      newChatFormVisible: false,
+      email: null,
+      chats: [],
+    };
+  }
+
+  selectChat = (chatIndex) => {
+    this.setState({ selectedChat: chatIndex });
+  };
+
+  newChatBtnClicked = () => {
+    this.setState({ newChatFormVisible: true, selectedChat: null });
+  };
+
+  componentDidMount = () => {
+    firebase.auth().onAuthStateChanged(async (_usr) => {
+      if (!_usr) {
+        this.props.history.push("/login");
+      } else {
+        await firebase
+          .firestore()
+          .collection("chats")
+          .where("users", "array-contains", _usr.email)
+          .onSnapshot(async (res) => {
+            const chats = res.docs.map((_doc) => _doc.data());
+            await this.setState({
+              email: _usr.email,
+              chats: chats,
+            });
+            console.log(this.state);
+          });
+      }
+    });
+  };
+
+  signOut = () => firebase.auth().signOut();
+
+  render() {
+    const { classes } = this.props;
+
+    return (
+      <div>
```

Comment:
FYI you might not need this wrapping `div`, you could replace this div with a fragment. I generally try to avoid adding extra divs to keep the DOM as flat as possible.

The syntax for a React fragment is

```
<React.Fragment>
   ...
</React.Fragment>
```
(...which I've seen you use elsewhere in your code already).

or, the same thing, even smaller;
```
<>
   ...
</>
```

Reply:
I learnt how to use the React fragment a little ago but forgot to use it here. I removed the `div` and put `<>...</>`